Assets/Plugins/Source/Editor/Windows/CreatePackageWindow.cs

```diff
@@ -203,22 +212,41 @@ protected override void RenderWindow()
                     labelSize.x,
                     labelSize.y);
 
-                EditorGUI.ProgressBar(progressBarRect, _progress, "");
+                lock (_progressLock)
+                {
+                    _displayedProgress = Mathf.Lerp(_displayedProgress, _actualProgress, Time.deltaTime * 2);
+                }
+
+                EditorGUI.ProgressBar(progressBarRect, _displayedProgress, "");
 
                 GUI.Label(labelRect, _progressText, customLabelStyle);
 
                 GUILayout.Space(20f);
                 if (GUILayout.Button("Cancel"))
                 {
-                    FileUtility.CleanDirectory(_packagingConfig.pathToOutput);
-                    _progress = 0.0f;
+                    _actualProgress = 0.0f;
                     _progressText = "";
                     _createPackageCancellationTokenSource?.Cancel();
+                    FileUtility.CleanDirectory(_packagingConfig.pathToOutput);
                 }
                 GUILayout.EndVertical();
             }
         }
 
+        private void UpdateProgressBar(CancellationToken cancellationToken)
```

Comment:
The name of this function looks like it would actually update the progress bar, but all it seems to do is wait until cancellation is requested.

Does this thread close itself out upon a completion / non-cancellation?

Reply:
It does indeed. I also agree that the function name is a bit misleading 🤔